services/cloud/availabilityzone/availabilityzone.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache v2.0 License.
+
+package availabilityzone
+
+import (
+	"github.com/microsoft/moc-sdk-for-go/services/cloud"
+	"github.com/microsoft/moc/pkg/errors"
+	"github.com/microsoft/moc/pkg/status"
+	wssdcloudcompute "github.com/microsoft/moc/rpc/cloudagent/cloud"
+)
+
+// this is only used in create/update
+func getRpcAvailabilityZone(s *cloud.AvailabilityZone) (*wssdcloudcompute.AvailabilityZone, error) {
+	if s == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "Availability zone object is nil")
+	}
+
+	if s.Name == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "AvailabilityZone object Name is empty")
+	}
+
+	availabilityZone := &wssdcloudcompute.AvailabilityZone{
+		Name:                     *s.Name,
+		Status:                   status.GetFromStatuses(s.Statuses),
+		LocationName:             *s.Location,
```

Comment:
Since nil check is done for 'Name', should we do nil check on other parameters too ? Ex: location.

Reply:
location is a required parameter, if the location is not provided the execution will not even come to this place. precheck for location is already added in the method `getAvailabilityZoneRequest` in wssd.go and after that precheck is done, then  `getRpcAvailabilityZone`  (i.e. the code above is called). so i dont think adding a precheck for location makes any difference here